native/screens/Applicants/Applicants.jsx

```diff
@@ -3,45 +3,60 @@ import { useEffect, useState } from 'react';
 import { useSelector } from 'react-redux';
 import uuid from 'react-native-uuid';
 import { handleToggle } from '../../services/utils/handlers';
+import { getOneProject } from '../../services/api/projects';
 
-export const Applicants = ({ navigation }) => {
+export const Applicants = ({ navigation, route }) => {
   const soloProject = useSelector((state) => state.projects.allProjects[0]);
-  const reduxUserID = useSelector((state) => state.ui.user._id);
+  const [currProject, setCurrProject] = useState(null);
   const [engineers, setEngineers] = useState([]);
   const [designers, setDesigners] = useState([]);
-  const [verifiedOwner, toggleVerifiedOwner] = useState(false);
 
   useEffect(() => {
-    if (soloProject && soloProject.owner === reduxUserID) {
-      toggleVerifiedOwner(true);
-    } else {
-      toggleVerifiedOwner(false);
-      return;
-    }
+    const setProject = async () => {
+      let fetchedProject;
 
-    const filterRoles = (role) => {
-      return soloProject.interested_applicants?.filter((applicant) => applicant.role === role);
-    };
+      if (!route.params) {
+        fetchedProject = soloProject;
+      } else {
+        fetchedProject = await getOneProject(route.params.projectID); // must be tested
+      }
+
+      const filterRoles = (role) => {
+        return fetchedProject.interested_applicants?.filter((applicant) => applicant.role === role);
+      };
 
-    setEngineers(() => filterRoles('Software Engineer'));
-    setDesigners(() => filterRoles('UX Designer'));
-  }, [reduxUserID, soloProject]);
+      setCurrProject(fetchedProject);
+      setEngineers(() => filterRoles('Software Engineer'));
+      setDesigners(() => filterRoles('UX Designer'));
+    };
+    setProject();
+  }, [route.params]);
```

Comment:
Does having route.params as the second argument here take care of making sure the screen re-renders when a user is routed here?

Reply:
Correct! Even though the `route.params` ends up being an identical object, React probably creates a new object in memory every time, forcing it to think new values have been passed. It's the only workaround i've discovered so far.